src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js

```diff
@@ -0,0 +1,138 @@
+      const data = await api.getActivationKey(activationKey);
+      setRole(data.role);
+      setServiceLevel(data.serviceLevel);
+      setUsage(data.usage);
+      setRepositories(data.additionalRepositories);
+    };
+    fetchKeyInformation();
+  }, []);
+
+  return (
+    <>
+      <TextContent>
+        <TextList component={TextListVariants.dl}>
+          <TextListItem component={TextListItemVariants.dt}>Name:</TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {activationKey}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>Role:</TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {role || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>SLA:</TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {serviceLevel || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>
+            Usage:
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {usage || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>
+            Additional <br /> repositories:
+            <Popover
+              bodyContent={
+                <TextContent>
+                  <Text>
+                    The core repositories for your operating system version are
+                    always enabled and do not need to be explicitally added to
+                    the activation key.
+                  </Text>
+                </TextContent>
+              }
+            >
+              <Button
+                variant="plain"
+                aria-label="About additional repositories"
+                className="pf-u-pl-sm pf-u-pt-0 pf-u-pb-0"
+                isSmall
+              >
+                <HelpIcon />
+              </Button>
+            </Popover>
+          </TextListItem>
+          <TextListItem
+            component={TextListItemVariants.dd}
+            className="pf-u-display-flex pf-u-align-items-flex-end"
+          >
+            {additionalRepositories?.length === 0 ? (
+              'None'
+            ) : (
+              <Popover
+                bodyContent={
+                  <TextContent>
+                    <Text component={TextVariants.h3}>
+                      Additional repositories
+                    </Text>
+                    <TableComposable
+                      aria-label="Additional repositories table"
+                      variant="compact"
+                    >
+                      <Thead>
+                        <Tr>
+                          <Th>Name</Th>
+                        </Tr>
+                      </Thead>
+                      <Tbody data-testid="additional-repositories-table">
+                        {additionalRepositories?.map((r, ri) => (
```

Comment:
This is personal preference but maybe call these (repo, index)? For me, there is some mental overhead required to convert `r` to `repo`. But we don't have a style guide that specifies either way and this is also fine if you prefer it.

Reply:
Fair, updated for better readability :)